### ISSUE TITLE
PID - ORCID, WikiData - Osoby

### ISSUE THREAD

**Issue by motyc** [ARUP-CAS/aiscr-webamcr#2594]
Cílem je doplnit ID podle ORCID a WikiData k těm osobám, které je mají přidělené (děje se zcela nezávisle na AMČR).

- [x] Doplnit pole:
-- `osoba.orcid`
-- `osoba.wikidata`
- [x] Do [amcr.xsd](https://github.com/ARUP-CAS/aiscr-webamcr/blob/dev/webclient/xml_generator/definitions/amcr.xsd) budou za element `osoba/rodne_prijmeni` doplněny elementy:
-- `<xs:element name="orcid" minOccurs="0" maxOccurs="1" type="xs:string"/> <!-- "{orcid}" -->`
-- `<xs:element name="wikidata" minOccurs="0" maxOccurs="1" type="xs:string"/> <!-- "{wikidata}" -->`
- [x] Editaci polí umožnit ve správě osob v administraci a při přidání nové osoby (přes modal u formulářů).
- [x] Pokud někdo zadá bez prefix, vždy doplnit:
--  `0000-0001-6592-6291` -> `https://orcid.org/0000-0001-6592-6291`
-- `Q59162343` -> `https://www.wikidata.org/entity/Q59162343`
- [x] Prověřit možnost našeptávání na základě API a automatickou validaci (nepovolit zápis neplatného ID):
-- https://info.orcid.org/documentation/api-tutorials/api-tutorial-read-data-on-a-record/
-- https://www.wikidata.org/wiki/Wikidata:REST_API

**Comment by pesikj** [comment]
A nebylo by dobré ještě dát možnost uživatelům, aby si to editovali ve svém profilu?

**Comment by motyc** [comment]
@pesikj Ano, ale to je v #2593, toto se týká hesláře osob, který spravujeme my. Tam by uživatelé mají možnost přidávat nová hesla, ale nikoli je zpětně editovat.

**Comment by motyc** [comment]
### Test v1.1.0-alpha-3
- [ ] Doplnit unique constraint na na pole orcid a wikidata
- [ ] Při zápisu přes formulář nová osoba je třeba akceptovat i null hodnoty.
- [ ] Při zápisu přes formulář nová osoba chybí našeptávač, jako je v administraci

**Comment by pesikj** [comment]
@motyc Na testu to vyhledávání v admin formuláři funguje

![Image](https://github.com/user-attachments/assets/9010ef29-522f-44a6-995e-981085327a3c)

**Comment by motyc** [comment]
@motyc Máš pravdu, teď mi taky hledání jde (i když by teda bylo lepší, kdyby to nebylo case sensitive). Bylo to ale o verzi zpět, tak nevím, zda to neovlivnila jiná úprava.

**Comment by pesikj** [comment]
@motyc Case insensitive jsem zkoušel, ale zatím mi to vychází jako o dost pomalejší.

**Comment by motyc** [comment]
@pesikj Ok no, tak to v tomto smyslu zatím nech a případně se k tomu vrátíme, až bude víc klidu se v tom hrabat a uvidíme to v praxi.

**Comment by pesikj** [comment]
@pesikj Tak to asi půjde, když se s query vymaže diakritika.

**Comment by motyc** [comment]
Super!

**Comment by motyc** [comment]
### Test v1.1.0-alpha-9
- [ ] https://amcr-test.aiscr.cz/admin/uzivatel/osoba/29122/change/
-- když chci nastavit null do pole Wikidata, systém mě zastaví, že nejsou data unikátní
-- pole jsou označená jako povinná, i když nejsou
- [ ] když do pole Wikidata skutečně zadám duplicitu, dostanu 500: http://192.168.254.30:5601/app/discover#/doc/44e15cc6-98d1-4cc3-815f-27089e09e9d1/amcr?id=YrTyZ5UBRGZYzcOY8u2y
- [ ] Wikidata - když zadám "David Novák" nic nenajde (bez ohledu formát); když jen "Novák", nabídne mi pouze "František Antonín Novák", tj. patrně jen první hodnotu?
- [ ] Pro ORCID stále chybí prefix (viz metadata pro https://amcr-test.aiscr.cz/admin/uzivatel/osoba/33955/change/)
- [ ] Modal formulář přidání osoby - našeptávače jsou rozbité, nelze testovat. Stále platné - pokud je s tím dramatický problém, lze to zatím odebrat a doplníme to samostatně po pořádném otestování. Stejně to musí primárně spravovat administrátoři.

**Comment by pesikj** [comment]
@motyc Našeptávač mi funguje (u dokumentů i externích zdrojů).

<img width="1403" alt="Image" src="https://github.com/user-attachments/assets/80cee954-925a-4d47-9592-60100aaaca04" />

**Comment by motyc** [comment]
@pesikj Ať to zkouším jakkoli, prostě je to celé jakoby zaseklé a nejde mi do pole psát:

![Image](https://github.com/user-attachments/assets/001810cd-af67-4d2b-b247-4cb9210769a4)

Zkoušel jsem Chrome i Firefox

**Comment by pesikj** [comment]
@motyc Tak to možná nefunguje jen pro přidávání nových dokumentů, ale v editaci by to fungovat mělo.

**Comment by motyc** [comment]
@pesikj Máš pravdu. Chce to tedy ale projít důsledně. Nefunguje to ani v akci pro vedoucího a asi budou i jiná místa. Prosím o kontrolu, kde všude to je a opravu všude.

**Comment by motyc** [comment]
### Test v1.1.0-alpha-10
- [ ] Vše ok, kromě těch problém s modalem, např. pro:
-- https://amcr-test.aiscr.cz/dokument/zapsat
-- https://amcr-test.aiscr.cz/arch-z/akce/zapsat
-- https://amcr-test.aiscr.cz/projekt/edit/X-C-000001089
-- https://amcr-test.aiscr.cz/dokument/model/zapsat

Možná došlo k nějaké chybě při slučování větví, protože ten [zde](https://github.com/ARUP-CAS/aiscr-webamcr/pull/2849#issuecomment-2706189602) referovaný [commit](https://github.com/ARUP-CAS/aiscr-webamcr/pull/2849/commits/b8187eb372c7a1774e0fd1a351b0eadfdb336cef) je jen integrace změn z test do [bugfix/2594](https://github.com/ARUP-CAS/aiscr-webamcr/pull/2849).

**Comment by motyc** [comment]
### Test v1.1.0-alpha-12

- [ ] Už se blížíme, ale opravdu je třeba cíleně vyhledat a odchytat všechna místa, kde je to použité. Tohle náhodné hledání na mé straně fakt není ideální. Ještě to nejede tu:
-- https://amcr-test.aiscr.cz/projekt/detail/C-202301502 přihlásit projekt - modal: ![Image](https://github.com/user-attachments/assets/4c2c75a7-ef7d-4d92-b501-7e6b0eb9b52c)
-- https://amcr-test.aiscr.cz/pas/zapsat
-- https://amcr-test.aiscr.cz/pas/edit/M-202300130-N00362
-- https://amcr-test.aiscr.cz/arch-z/akce/zapsat - Další vedoucí (tabulka dole)
-- https://amcr-test.aiscr.cz/dokument/detail/C-TX-192600929/cast/C-TX-192600929-D001 - editace neident akce
- [ ] Také jsem si teď všiml, že chybí heslar.forms.OsobaAdminForm.wikidata.tooltip (ve formuláři vůbec není, asi opomenuto)

**Comment by pesikj** [comment]
@motyc U toho přepínání z modalu do modalu to standardní řešení nefunguje, to se bude muset řešit zvlášť.

**Comment by motyc** [comment]
@pesikj A mám tedy udělat zvlášť issue, nebo to ještě doplníš sem?

**Comment by pesikj** [comment]
@motyc Můžeme to doplnit sem.
